aep/general/0008/aep.md

```diff
@@ -1,5 +1,259 @@
-# AIP Style guide
+# AEP Style and Guidance
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/32) for more
-information.
+AEP stands for **API Enhancement Proposal**, which is a design document
+providing high-level, concise documentation for API design and development. The
+goal is for these documents to serve as the source of truth for API-related
+documentation and the way API teams discuss and come to consensus on
+API guidance.
+
+AEPs are most useful when they are clear and concise, and cover a single topic
+or inquiry well. In the same way that AEPs describe consistent patterns and
+style for use in APIs, they also _follow_ consistent patterns and style.
+
+## Guidance
+
+- AEPs **must** cover a single, discrete topic, and provide clear, actionable
+  guidance.
+- AEPs **must not** duplicate or contradict guidance in another AEP.
+- AEPs **may** also cover what _not_ to do, but **should not** cover _only_
+  anti-patterns.
+- If AEP guidance is conditional (e.g. a design pattern such as Jobs), the
+  guidance **must** clearly explain under what conditions the guidance should be
+  followed.
+
+Guidance contained within an AEP **must** be beneficial to one or more
+types of clients or their authors, including but not limited to:
+
+- Asset inventories which can be used to audit and analyze resources.
+- Command line interfaces for exploration and simple automation.
+- Custom controllers (e.g. auto-scalers) which poll live state and adjust
+  resource configuration accordingly.
+- [IaC][] clients for orchestration and automation of multiple resources.
+- Recommendation tools which provide guidance on which APIs are useful for
+  specific use cases, and how to use them.
+- SDKs to interact with an API from a programming language, often used heavily
+  for data-plane operations.
+- Security orchestration, automation, and remediation tools.
+- Simple scripts to automate or orchestrate tasks.
+- Test frameworks.
+- Tools that operate on resource data at rest.
+- Visual User Interfaces for visualization and one-off manual actions.
+- Users.
+
+Examples of enhancements include:
+
+- Requiring new proto annotations that enable more descriptive interfaces on
+  clients (e.g. requiring `singular` and `plural` on a `library.api.resource`
```

Comment:
```suggestion
  clients (e.g. requiring `singular` and `plural` on a `google.api.resource`
```

Reply:
I thought we decided to change all the "google" references to "library". Did we reverse that decision?

Reply:
fixed.